Docs-10014-PM2.6NewFeatures_ESB (#112)
mulesoft/docs-partner-manager: modules/ROOT/pages/endpoint-as2-receive.adoc

= AS2 Receive Endpoint Settings

Your organization's AS2 endpoint settings determine where to receive inbound data from partners.
One AS2 endpoint can receive inbound data from multiple partners.

Create AS2 identifiers for your organization as host and for your partners in their respective xref:configure-host.adoc[host] and xref:configure-partner.adoc[partner] profile pages.

[%header%autowidth.spread]
|===
| Setting | Description | Required
| Endpoint Name

> | The automatically-generated name that identifies the endpoint, which you can subsequently modify manually.

Reviewer: no hyphen; see "hyphens" in style ref.

Reviewer 2: Done!

|===

== AS2 Receive Endpoint URL

> === Deploy to CloudHub on Public Port

Reviewer: on > on a

Reviewer 2: Done!

When an AS2 receive endpoint is deployed with the private port configuration enabled, the endpoint application is deployed on the public HTTPS port (8082), with the traffic routed through the xref:runtime-manager::dedicated-load-balancer-tutorial#shared-load-balancers [shared load balancer]. The value in the URL field is automatically generated when you deploy a message flow to CloudHub that uses this endpoint.

> You can view this URL by selecting the receiving endpoint from your deployed message flow or by clicking the endpoint name from the endpoints list. You can share this URL with the partners that would send AS2 messages to this endpoint.

Reviewer: Do we need "would"?

Reviewer 2: NOPE we do not. Removed.

> If your organization uses CloudHub VPC without a dedicated load balancer, your organization’s platform administrator can configure the necessary xref:runtime-manager::vpc-firewall-rules-concept.adoc[VPC Firewall Rules] to allow specific IP addresses of your

Reviewer: ```suggestion
If your organization uses CloudHub VPC without a dedicated load balancer, your organization’s platform administrator can configure the necessary xref:runtime-manager::vpc-firewall-rules-concept.adoc[VPC firewall rules] to allow specific IP addresses of your
```

partners to access the endpoint.

> === Deploy to CloudHub on Private Port

Reviewer: ```suggestion
=== Deploy to CloudHub on a Private Port
```

When an AS2 receive endpoint is deployed with the private port configuration enabled, the endpoint is deployed to the private HTTPS port (8092) and the internal URL of the endpoint appears on the *Message Flow* or *Endpoint* detail page.

> After deploying the first message flow using the AS2 receive endpoint configuration, you can view the runtime application name for the endpoint from the *Endpoint* details page or the *Receive* section of your message flow. To configure xref:runtime-manager/lb-mapping-rules[URL mapping rules] in the xref:runtime-manager::cloudhub-dedicated-load-balancer.adoc[Dedicated Load Balancer] to forward AS2 messages received from your partners to the application, provide the application name to your organization’s platform administrator.

Reviewer: ```suggestion
After deploying the first message flow using the AS2 receive endpoint configuration, you can view the runtime application name for the endpoint from the *Endpoint* details page or the *Receive* section of your message flow. To configure xref:runtime-manager/lb-mapping-rules[URL mapping rules] in the xref:runtime-manager::cloudhub-dedicated-load-balancer.adoc[dedicated load balancer] to forward AS2 messages received from your partners to the application, provide the application name to your organization’s platform administrator.
```

> Your organization’s platform administrator can also xref:runtime-manager::lb-whitelists.adoc[whitelist] your partners’ IP addresses in the DLB settings to allow specific IP addresses to access the endpoint.

Reviewer: We really can't change the text in [ ] but STET the xref syntax?

xref:runtime-manager::lb-whitelists.adoc[add to a list of allowed addresses]

Reviewer 2: Done - modified slightly to fit the context of the sentence

> Your organization’s platform administrator can also xref:runtime-manager::lb-whitelists.adoc[whitelist] your partners’ IP addresses in the Dedicated Load Balancer settings to allow specific IP addresses to access the endpoint.

Reviewer: Reword to not use "whitelist." See style ref. entries for inclusive/noninclusive language.

Reviewer: GLOBALLY lowercase "dedicated load balancer," although you can use DLB after introducing the abbr.

Reviewer 2: Done!

Reviewer 2: I would love to change "whitelist", but that is the actual name of the [topic in Runtime Manager](https://docs.mulesoft.com/runtime-manager/lb-whitelists).

> The external URL that you will share with your partners will be https://{DLB-domain}/{Input-path}/receive-as2

Reviewer: Avoid future tense. Also period.

Reviewer 2: Done

> For example, if the name of your DLB domain is `mythical.lb.anypointdns.net`, and the runtime application name is `b2b-inbound-as2-zap3`, then your administrator will add a URL mapping rule in the Dedicated Load Balancer settings to forward incoming requests to the runtime application within your VPC.

Reviewer: No need for second comma. Ditto future tense, DLB:
```suggestion
For example, if the name of your DLB domain is `mythical.lb.anypointdns.net` and the runtime application name is `b2b-inbound-as2-zap3`, then your administrator adds a URL mapping rule in the DLB settings to forward incoming requests to the runtime application within your VPC:
```
Colon at end of text preceding image

image::URL-mapping-rules.png[URL mapping rules in the DLB]

> With these configurations, the external AS2 endpoint URL is `https://mythical.lb.anypointdns.net/as2/mythical/receive-as2/``.

Reviewer: ```suggestion
With these configurations, the external AS2 endpoint URL is `https://mythical.lb.anypointdns.net/as2/mythical/receive-as2/`.
```

> With these configurations, the external AS2 endpoint URL is https://mythical.lb.anypointdns.net/as2/mythical/receive-as2/.

Reviewer: URLs take mono font.

Reviewer 2: Fixed!

> === Deploy to On-Premise Mule Runtime

Reviewer: ```suggestion
=== Deploy to On-Premises Mule Instance
```

> If you are deploying to an on-premises Mule instance with a firewall and load balancer, work with your organization's hosting or network security team to obtain the externally accessible DNS or the name of the instance host. You must replace the `{runtime_host}` value in the URL with the specific value for your organization. For example, if the DNS host name is `b2b.mycompany.com`, the AS2 receiver URL is `+https://b2b.mycompany.com/receive-as2/+`.

Reviewer: ```suggestion
If you want to deploy to an on-premises Mule instance that uses a firewall and load balancer, work with your organization's hosting or network security team to obtain the externally accessible DNS or the name of the instance host. You must replace the `{runtime_host}` value in the URL with the specific value for your organization. For example, if the DNS host name is `b2b.mycompany.com`, then the AS2 receiver URL is `+https://b2b.mycompany.com/receive-as2/+`.
```
